tests/test_marshmallow.py

```diff
@@ -0,0 +1,188 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Test marshmallow fields and schema."""
+
+
+from dataclasses import dataclass, field
+from typing import Any, cast
+
+from marshmallow_dataclass import class_schema
+
+from frequenz.quantities import Energy, Percentage, Power, Temperature, Voltage
+from frequenz.quantities.marshmallow import QuantitySchema
+
+
+@dataclass
+class Config:
+    """Configuration test class."""
+
+    my_percent_field: Percentage = field(
+        default_factory=lambda: Percentage.from_percent(25.0),
+        metadata={
+            "metadata": {
+                "description": "A percentage field",
+            },
+        },
+    )
+
+    my_power_field: Power = field(
+        default_factory=lambda: Power.from_watts(100.0),
+        metadata={
+            "metadata": {
+                "description": "A power field",
+            },
+        },
+    )
+
+    my_energy_field: Energy = field(
+        default_factory=lambda: Energy.from_watt_hours(100.0),
+        metadata={
+            "metadata": {
+                "description": "An energy field",
+            },
+        },
+    )
+
+    voltage_always_string: Voltage = field(
+        default_factory=lambda: Voltage.from_kilovolts(200.0),
+        metadata={
+            "metadata": {
+                "description": "A voltage field that is always serialized as a string",
+                "serialize_as_string": True,
+            },
+        },
+    )
+
+    temp_never_string: Temperature = field(
+        default_factory=lambda: Temperature.from_celsius(100.0),
+        metadata={
+            "metadata": {
+                "description": "A temperature field that is never serialized as a string",
+                "serialize_as_string": False,
+            },
+        },
+    )
+
+    @classmethod
+    def load(cls, config: dict[str, Any]) -> "Config":
+        """Load the configuration."""
+        schema = class_schema(cls, base_schema=QuantitySchema)()
+        return cast(Config, schema.load(config))
```

Comment:
I think the class method can return `Self` (Self imported from typing)
```suggestion
    @classmethod
    def load(cls, config: dict[str, Any]) -> Self:
        """Load the configuration."""
        schema = class_schema(cls, base_schema=QuantitySchema)()
        return cast(Self, schema.load(config))
```